portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,24 +35,38 @@
     
     public class Name 
     {
-        private ArrayList<String> names = new ArrayList<String>();
+        public ArrayList<String> names = new ArrayList<String>();
     }
   
-    private Name name = new Name();
+    Name name = new Name();
+   
 
-  
+    
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Name name = new Name();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Gson gson = new Gson(); 
+        Query query = new Query("Names");
+        PreparedQuery results = datastore.prepare(query);
+        
+        for (Entity entity : results.asIterable()) 
+        {
+            name.names.add( (String) entity.getProperty("viewer0"));
+
+        }
         String json = gson.toJson(name);
         response.setContentType("application/json;");
         response.getWriter().println(json);
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        getNames(request);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity namesEntity = new Entity("Names");
+        getNames(request, namesEntity, datastore);
+
         response.sendRedirect("https://8080-dot-10831547-dot-devshell.appspot.com/");
```

Comment:
I think this should be a relative path. Does `response.sendRedirect(".");` work?